live-examples/js-examples/expressions-functionasteriskexpression.html

```diff
@@ -0,0 +1,13 @@
+<pre>
+<code id="static-js">function* foo() {
+  yield 'a';
+  yield 'b';
+  yield 'c';
+}
+
+const [...values] = foo();
```

Comment:
Nit: To me this looks a bit complex. Maybe something like:

```
function* foo() {
  yield 'a';
  yield 'b';
  yield 'c';
}

for (let val of foo()) {
  console.log(val);
}

// expected output: "a","b","c"
```

would be easier to grasp?

Reply:
Yes we could do this. Note that I've been avoiding console.logs within loops up to now. Maybe I shouldn't? It's just the output will be on separate lines. ie

```
> "a"
> "b"
> "c"
```

and I'm not sure how to represent that neatly.

Reply:
> So I'd probably be tempted to do :
...

Yes, I think this would be OK.

I do take your point that " It's just the output will be on separate lines...and I'm not sure how to represent that neatly." But I think, realistically that if we say:

`// expected output: "a","b","c"`

and they get:

```
> "a"
> "b"
> "c"
```
...then most people will understand this. Maybe not, though :).